forc-plugins/forc-mcp/src/forc_call/mod.rs

```diff
@@ -0,0 +1,1414 @@
+mod resources;
+
+use crate::McpToolModule;
+use resources::{COMMON_COMMANDS_URI, CONTRACT_SAMPLES_URI, TYPE_ENCODING_REFERENCE_URI};
+use rmcp::{
+    handler::server::{router::tool::ToolRouter, tool::Parameters},
+    model::*,
+    schemars::{self, JsonSchema},
+    service::RequestContext,
+    tool, tool_handler, tool_router, Error as McpError, RoleServer, ServerHandler,
+};
+use serde::Deserialize;
+use serde_json::Value;
+use std::{collections::HashMap, future::Future, pin::Pin, str::FromStr};
+
+#[derive(Debug, Deserialize, JsonSchema)]
+pub struct CallContractArgs {
+    pub contract_id: String,
+    pub abi: String,
+    pub function: String,
+    #[serde(default)]
+    pub function_args: Vec<String>,
+    #[serde(default = "default_mode")]
+    pub mode: String,
+    pub node_url: Option<String>,
+    pub signing_key: Option<String>,
+    #[serde(default)]
+    pub amount: u64,
+    pub asset_id: Option<String>,
+    pub gas_price: Option<u64>,
+    #[serde(default)]
+    pub verbosity: u8,
+}
+
+#[derive(Debug, Deserialize, JsonSchema)]
+pub struct ListFunctionsArgs {
+    pub contract_id: String,
+    pub abi: String,
+}
+
+#[derive(Debug, Deserialize, JsonSchema)]
+pub struct TransferAssetsArgs {
+    pub signing_key: String,
+    pub recipient: String,
+    pub amount: u64,
+    pub asset_id: Option<String>,
+    pub node_url: Option<String>,
+    #[serde(default)]
+    pub verbosity: u8,
+}
+
+#[derive(Debug, Deserialize, JsonSchema)]
+pub struct GetExecutionTraceArgs {
+    pub trace_events: Vec<HashMap<String, Value>>, // JSON objects representing TraceEvent
```

Comment:
Do we want to make this a doc comment?

Reply:
https://github.com/FuelLabs/sway/pull/7284/commits/b796f80130163f0644ec9c63f3cc1d6ccf06ff32